src/components/register/register.jsx

```diff
@@ -0,0 +1,247 @@
+/* eslint-disable react/no-children-prop */
+import React, { useState } from 'react';
+import { instanceOf } from 'prop-types';
+import { useNavigate } from 'react-router-dom';
+import {
+  FormControl,
+  FormLabel,
+  InputGroup,
+  Input,
+  Heading,
+  InputRightAddon,
+  Button,
+  Flex,
+  Spacer,
+  VStack,
+  Center,
+  Text,
+} from '@chakra-ui/react';
+import { Cookies, withCookies } from '../../utils/cookie_utils';
+import { registerWithEmailAndPassword, signInWithGoogle } from '../../utils/auth_utils';
+
+const Register = ({ cookies }) => {
+  const [errorMessage, setErrorMessage] = useState();
+  const [firstName, setFirstName] = useState();
+  const [lastName, setLastName] = useState();
+  const [email, setEmail] = useState();
+  const [password, setPassword] = useState();
+  const [checkPassword, setCheckPassword] = useState();
+  const [showPassword, setShowPassword] = useState();
+  const [showCheckPassword, setShowCheckPassword] = useState();
+  const [role, setRole] = useState();
+  const navigate = useNavigate();
+
+  const handleSubmit = async e => {
+    e.preventDefault();
+    try {
+      if (password !== checkPassword) {
+        throw new Error("Passwords don't match");
+      }
+      await registerWithEmailAndPassword(email, password, firstName, lastName, role, navigate, '/');
+    } catch (error) {
+      setErrorMessage(error.message);
+    }
+  };
+
+  /**
+   * This function handles signing up with Google
+   * If the user logs in and is new, they are directed to a new-user path
+   * If the user logs in and isn't new, they are directed to the dashboard.
+   * If the user fails to log in, they are directed back to the login screen
+   */
+  const handleGoogleSignIn = async e => {
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'handleGoogleSignIn' is assigned a value but never used.